src/main/java/com/fcgl/madrid/points/service/UserActionService.java

```diff
@@ -1,22 +1,70 @@
 package com.fcgl.madrid.points.service;
 
+import com.fcgl.madrid.points.model.Action;
+import com.fcgl.madrid.points.model.Trophy;
 import com.fcgl.madrid.points.model.UserAction;
+import com.fcgl.madrid.points.model.UserTrophy;
+import com.fcgl.madrid.points.repository.ActionRepository;
+import com.fcgl.madrid.points.repository.TrophyRepository;
 import com.fcgl.madrid.points.repository.UserActionRepository;
+import com.fcgl.madrid.points.repository.UserTrophyRepository;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
 
 import java.util.List;
 
 @Service
 public class UserActionService {
     private UserActionRepository userActionRepository;
+    private ActionRepository actionRepository; // Is this the correct way to access the action table?
```

Comment:
I have the feeling that this is wrong but I couldn't figure out a better way. would appreciate suggestions

Reply:
Yes, you place it here so that it's accessible. Then initialize it in the constructor as you did below. The `@Autowired` does the work. Not sure if there is a better way? That's how other teams have been doing it as well